src/canopy/llm/openai.py

```diff
@@ -175,15 +197,22 @@ def enforced_function_call(self,
         function_dict = cast(ChatCompletionToolParam,
                              {"type": "function", "function": function.dict()})
 
-        chat_completion = self._client.chat.completions.create(
-            messages=[m.dict() for m in messages],
-            model=self.model_name,
-            tools=[function_dict],
-            tool_choice={"type": "function",
-                         "function": {"name": function.name}},
-            max_tokens=max_tokens,
-            **model_params_dict
-        )
+        try:
+            chat_completion = self._client.chat.completions.create(
+                messages=[m.dict() for m in messages],
+                model=self.model_name,
+                tools=[function_dict],
+                tool_choice={"type": "function",
+                             "function": {"name": function.name}},
+                max_tokens=max_tokens,
+                **model_params_dict
+            )
+        except openai.OpenAIError as e:
+            provider_name = self.__class__.__name__.replace("LLM", "")
+            raise RuntimeError(
+                f"Failed to use {provider_name}'s {self.model_name} model for chat completion.\n"
```

Comment:
I think it's better to have this message indicate that the error is coming from the function calling method and not for the chat completion method, it will be easier for debugging